fiaas_deploy_daemon/deployer/kubernetes/deployment/deployer.py

```diff
@@ -82,7 +83,7 @@ def deploy(self, app_spec, selector, labels, besteffort_qos_is_required):
 
         automount_service_account_token = app_spec.admin_access
         init_containers = []
-        service_account_name = "default"
+        service_account_name = app_spec.name if self._enable_service_account_per_app is True else "default"
```

Comment:
Does this cause the service-account to be created if it doesn't exist?

Reply:
No I don't think so. Creating the service account is not part of this PR

Reply:
Ok. The discussion in the previous PR seemed to arrive at the conclusion that's how we'd implement this. It could be an idea to have some content in the PR description explaining the context - i.e. link it to the issue, and if the implementation isn't matching the proposal, explain it.

Either way, the help text suggests that if the account doesn't exist then it will fallback to use default. Is that the kubernetes default behaviour, or will it fail the deploy?

Reply:
Thanks Greg. We haven't checked that situation but I expect it would fail. I can update the help text: it was written aspirationally at the beginning of the PR!

I agree, we need to give more context: we'll make sure that we match it with a GitHub issue so you see what we're aiming for. Though right now, it feels more like sculpture: we'll find out what the feature ends up like when it's done ;)

Reply:
I also wanted to shout out to @sa-mao and @PerGon as co-authors: we worked on this in a "mob session" over the last two weeks.

Reply:
Ah, ok. Sounds good.
Cc @cdemonchy as you folks were interested in this feature too